examples/all-clusters-app/telink/src/AppTask.cpp

```diff
@@ -108,6 +109,17 @@ using namespace ::chip::DeviceLayer::Internal;
 
 AppTask AppTask::sAppTask;
 
+class AppFabricTableDelegate : public FabricTable::Delegate
+{
+    void OnFabricRemoved(const FabricTable & fabricTable, FabricIndex fabricIndex)
+    {
+        if (chip::Server::GetInstance().GetFabricTable().FabricCount() == 0)
+        {
+            chip::Server::GetInstance().ScheduleFactoryReset();
```

Comment:
This does not look right.

If a commissioner deletes a single existing fabric over PASE during commissioning, before it has done AddNOC, doing a reset does not seem like the right behavior.  Is it?

Reply:
Could you explain the scenario to reproduce your situation?
During pairing I've never seen removing last fabric.
In our case this pease of code appears only when pairing is removed. So if latest pairing is removed we are doing factory reset.

Reply:
> Could you explain the scenario to reproduce your situation?

Say a device is commissioned into a single fabric.  That fabric opens a commissioning window.  A new commissioner connects via PASE, and before it does AddNOC removes the existing fabric.  Why, that's up to it; it just does.  Should that close the PASE session immediately and reset the device?

I just looked more carefully at the spec, and at least the "should it close the PASE session?" answer seems to be "yes".  So doing a reset here is probably fine.